### PR TITLE
Move GSProcessing docs to main repo documentation.

### PR DESCRIPTION
**Pull request by thvasilo**
*Issue #, if available:*

*Description of changes:*

Move the GSProcessing docs under the main repo to allow publishing under common readthedocs project.

Add a new "Distributed Processing" section at the index root, rename "Scale to Giant Graphs" to "Distributed Training" to differentiate between processing and training.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by zhjwy9343** [comment]
If we move these docs to the main /docs repo, are these make, conf.py files necessary. Can we remove them?

**Comment by thvasilo** [comment]
> If we move these docs to the main /docs repo, are these make, conf.py files necessary. Can we remove them?

This PR removes them.

**Comment by zhjwy9343** [comment]
When I run Sphinx `make html`. Get the following warning and errors. Could you take a look?

```
/data/theo/graphstorm/docs/source/gs-processing/developer/developer-guide.rst:37: WARNING: Title underline too short.

Install ``pyenv``
~~~~~~~~~~~~~
/data/theo/graphstorm/docs/source/gs-processing/developer/developer-guide.rst:37: WARNING: Title underline too short.

Install ``pyenv``
~~~~~~~~~~~~~
/data/theo/graphstorm/docs/source/gs-processing/developer/developer-guide.rst:75: WARNING: Title underline too short.

Install ``poetry``
~~~~~~~~~~~~~~
/data/theo/graphstorm/docs/source/gs-processing/developer/developer-guide.rst:75: WARNING: Title underline too short.

Install ``poetry``
~~~~~~~~~~~~~~
/data/theo/graphstorm/docs/source/gs-processing/developer/developer-guide.rst:85: WARNING: Title underline too short.

Install dependencies through ``poetry``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/theo/graphstorm/docs/source/gs-processing/developer/developer-guide.rst:85: WARNING: Title underline too short.

Install dependencies through ``poetry``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/theo/graphstorm/docs/source/gs-processing/developer/developer-guide.rst:180: WARNING: Title underline too short.

Use a pre-commit hook to ensure ``black`` and ``pylint`` runs before commits
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/data/theo/graphstorm/docs/source/scale/distributed.rst:4: WARNING: Duplicate explicit target name: "nfs tutorial".
/data/theo/graphstorm/docs/source/scale/distributed.rst:4: WARNING: Duplicate explicit target name: "nfs guide".
looking for now-outdated files... none found
pickling environment... done
checking consistency... /data/theo/graphstorm/docs/source/gs-processing/developer/developer-guide.rst: WARNING: document isn't included in any toctree
/data/theo/graphstorm/docs/source/gs-processing/developer/input-configuration.rst: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] tutorials/quick-start                                                                                                                                                                                                                                
/data/theo/graphstorm/docs/source/gs-processing/usage/amazon-sagemaker.rst:37: WARNING: unknown document: '/usage/distributed-processing-setup'
/data/theo/graphstorm/docs/source/gs-processing/usage/amazon-sagemaker.rst:54: WARNING: unknown document: '/usage/distributed-processing-setup'
/data/theo/graphstorm/docs/source/tutorials/own-data.rst:262: WARNING: Could not lex literal_block as "json". Highlighting skipped.
```

File: docs/source/gs-processing/gs-processing-getting-started.rst [hunk 1]
```diff
@@ -42,13 +33,37 @@ With PyEnv installed you can create and activate a Python 3.9 environment using
...
+.. code-block:: bash
+
+    # Ensure Python is at least 3.9
+    python -V
+    cd graphstorm/graphstorm-processing
+    pip install .
 
-Install Java 8, 11, or 17
-~~~~~~~~~~~~~~~~~~~~~~~~~
+Install using poetry
```
[the remark this answers is not in the document]
Comment (thvasilo) [reply docs/source/gs-processing/gs-processing-getting-started.rst:50]: It's meant to indicate how to install the library using `poetry`. I'll rephrase to make this more clear.

**Comment by zhjwy9343** [comment]
After compile the latest commit, I found there is a place like,
`For more info on pyen see its documentation. <https://github.com/pyenv/pyenv>`. 
But I cannot find this in the source code. Do you know where it is? And  fix this anchor format.

**Review by zhjwy9343 — APPROVED** [review]
LGTM

**Pull request merged**